### PR TITLE
Updated error responses and tests for RPC methods

### PR DESCRIPTION
**Pull request by jingsting**
## Purpose
Added more error responses for RPC methods

## Approach
- For `topl_confirmationStatus`: return "Not Found" when a transaction is not found in mempool and history, instead of returning an error
- For `admin_updateRewardsAddress`: return a confirmation instead of the requested address

## Testing
- Added test for invalid securityRoot(not base58 or invalid length)
- Added test for invalid assetCode
- Updated RPC spec for `topl_confirmationStatus`
- Added test for unlocking a non-existent keyfile with `admin_unlockKeyfile`
- Added test for the response from `admin_updateRewardsAddress `

## Tickets
#864 
#982 
#1032 
#1192 
#1040

### CODE REVIEW

File: topl-rpc/src/main/scala/co/topl/rpc/ToplRpcCodecs.scala [hunk 1]
```diff
@@ -119,6 +125,8 @@ trait TransactionRpcParamsEncoders extends SharedCodecs {
   implicit val transactionBroadcastTxParamsEncoder: Encoder[ToplRpc.Transaction.BroadcastTx.Params] =
     Encoder.forProduct1("tx")(_.tx)
 
+  implicit val transactionEncodeTransferParamsEncoder: Encoder[ToplRpc.Transaction.EncodeTransfer.Params] =
+    Encoder.forProduct1("unprovenTransaction")(_.unprovenTransaction)
```
Comment (b-vennes) [inline topl-rpc/src/main/scala/co/topl/rpc/ToplRpcCodecs.scala:129]: Are we able to use `deriveEncoder` instead of `Encoder.forProduct1`?

File: node/src/test/scala/co/topl/api/AdminRPCSpec.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        """.stripMargin)
+
+      httpPOST(requestBody) ~> route ~> check {
+        val res: Json = parse(responseAs[String]).value
+        val error = res.hcursor.downField("error").as[Json].toString
+        error should include("Cannot find a unique matching keyfile")
+      }
+    }
+
+    "Return return a confirmation after successfully updating the reward address" in {
```
Comment (b-vennes) [inline node/src/test/scala/co/topl/api/AdminRPCSpec.scala:33]: Possible typo here with `Return return`

File: node/src/test/scala/co/topl/api/NodeViewRPCSpec.scala [hunk 3]
```diff
@@ -157,17 +156,46 @@ class NodeViewRPCSpec extends AnyWordSpec with Matchers with RPCMockState with E
       view().mempool.remove(txs.head)
     }
 
+    "Return correct error response when an incorrect modifier id type is used" in {
```
Comment (b-vennes) [inline node/src/test/scala/co/topl/api/NodeViewRPCSpec.scala:159]: It might be good to break this out into multiple tests, since you are testing three different API endpoints

File: node/src/test/scala/co/topl/api/NodeViewRPCSpec.scala [hunk 4]
```diff
@@ -157,17 +156,46 @@ class NodeViewRPCSpec extends AnyWordSpec with Matchers with RPCMockState with E
...
+        }
+      }
+
+      httpPOST(requestBody("blockId", "topl_blockById", txId)) ~> route ~> check {
+        val res: String = parse(responseAs[String]).value.hcursor.downField("error").as[Json].toString
+        res should include("The requested id's type is not an id type for Block")
+      }
+    }
+
     "Return correct error response when an id with non-base58 character is used for querying transactions" in {
```
Comment (b-vennes) [inline node/src/test/scala/co/topl/api/NodeViewRPCSpec.scala:188]: We might also want to consider breaking this one out into multiple tests with each RPC function and id type.

File: node/src/test/scala/co/topl/api/NodeViewRPCSpec.scala (hunk 3 above)
Comment (b-vennes) [reply node/src/test/scala/co/topl/api/NodeViewRPCSpec.scala:159]: I see we are doing this a lot inside this Spec, it might be better to make it an issue instead of trying to fix it all here

**Review by b-vennes — COMMENTED** [review]
The changes to the `NodeViewRpcHandlerImpls` looks good. I like the new user-friendly error messages. I have a few comments on the added Json codes and tests.

File: node/src/test/scala/co/topl/api/AdminRPCSpec.scala (hunk 2 above)
Comment (jingsting) [reply node/src/test/scala/co/topl/api/AdminRPCSpec.scala]: Fixed that 😂

File: topl-rpc/src/main/scala/co/topl/rpc/ToplRpcCodecs.scala (hunk 1 above)
Comment (jingsting) [reply topl-rpc/src/main/scala/co/topl/rpc/ToplRpcCodecs.scala:129]: I just tried it but couldn't compile with this: could not find Lazy implicit value

**Comment by scasplte2** [comment]
Please make a ticket about removing the RpcClient and RpcServer @jingsting

File: node/src/main/scala/co/topl/rpc/handlers/AdminRpcHandlerImpls.scala [hunk 5]
```diff
@@ -71,7 +71,7 @@ class AdminRpcHandlerImpls(forgerInterface: ForgerInterface, keyManagerInterface
       keyManagerInterface
         .updateRewardsAddress(params.address)
         .leftMap(e => ToplRpcErrors.genericFailure(e.toString): RpcError)
-        .map(_ => ToplRpc.Admin.UpdateRewardsAddress.Response(params.address.toString))
+        .map(_ => ToplRpc.Admin.UpdateRewardsAddress.Response("Updated reward address"))
```
Comment (scasplte2) [inline node/src/main/scala/co/topl/rpc/handlers/AdminRpcHandlerImpls.scala:74]: why did you remove the address from this message? I think a user would be more comfortable if the response was `Updated reward address to ${params.address}`

File: node/src/test/scala/co/topl/api/AdminRPCSpec.scala [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        |     "address": "${keyRingCurve25519.addresses.head.toString}",
+        |     "password": "test"
+        |   }]
+        |}
+        """.stripMargin)
+
+      httpPOST(requestBody) ~> route ~> check {
+        val res: Json = parse(responseAs[String]).value
+        val error = res.hcursor.downField("error").as[Json].toString
+        error should include("Cannot find a unique matching keyfile")
```
Comment (scasplte2) [inline node/src/test/scala/co/topl/api/AdminRPCSpec.scala:29]: where is this error defined?

File: node/src/test/scala/co/topl/api/NodeViewRPCSpec.scala (hunk 3 above)
Comment (scasplte2) [reply node/src/test/scala/co/topl/api/NodeViewRPCSpec.scala:159]: I agree with Braden, we should break these into individual tests so that a unit test checks one thing (unless there is a good reason not to)

File: node/src/main/scala/co/topl/rpc/handlers/AdminRpcHandlerImpls.scala (hunk 5 above)
Comment (jingsting) [reply node/src/main/scala/co/topl/rpc/handlers/AdminRpcHandlerImpls.scala:74]: Just add  ${params.address} back in

File: node/src/test/scala/co/topl/api/AdminRPCSpec.scala (hunk 6 above)
Comment (jingsting) [reply node/src/test/scala/co/topl/api/AdminRPCSpec.scala:29]: It's in the `checkValid` method in KeyRing.scala. It gets thrown when the number of keyfile found with the address is not exactly one.

File: node/src/test/scala/co/topl/api/NodeViewRPCSpec.scala (hunk 3 above)
Comment (jingsting) [reply node/src/test/scala/co/topl/api/NodeViewRPCSpec.scala:159]: These API endpoints are being used together here because this test wants the error response to be the same when the id type is incorrect. But I'm ok with separating them into individual test too

Comment (jingsting) [reply node/src/test/scala/co/topl/api/NodeViewRPCSpec.scala:159]: Yeah, that makes sense to me

File: node/src/test/scala/co/topl/api/NodeViewRPCSpec.scala (hunk 4 above)
Comment (jingsting) [reply node/src/test/scala/co/topl/api/NodeViewRPCSpec.scala]: 👍  Separated these tests
